src/java/org/apache/cassandra/db/lifecycle/Tracker.java

```diff
@@ -270,6 +272,20 @@ public void updateInitialSSTableSize(Iterable<SSTableReader> sstables)
 
     public void addSSTables(Collection<SSTableReader> sstables)
     {
+        Preconditions.checkState(!cfstore.metadata().replicationType().isTracked());
+        addSSTablesInternal(sstables, false, true, true);
+    }
+
+    public void addSSTablesTracked(Collection<SSTableReader> sstables)
+    {
+        Preconditions.checkState(cfstore.metadata().replicationType().isTracked());
+        for (SSTableReader sstable : sstables)
+        {
+            ImmutableCoordinatorLogOffsets logOffsets = sstable.getCoordinatorLogOffsets();
+            Preconditions.checkState(logOffsets.mutations().isEmpty());
+            Preconditions.checkState(!logOffsets.transfers().isEmpty());
```

Comment:
Couple thoughts:

1.) Would `ImmutableCoordinatorLogOffsets` benefit from a method like `isTransfer()` that we could use here? ^
2.) I'd consider actual error messages here (especially after collapsing the two checks above with `isTransfer()`), although if we used assertions correctly, those would be better (given this is a programmer error if it happens).

Reply:
I'm not sure what isTransfer would represent, given that we'll have ICLO containing a mix of mutations and transfers after compaction has happened. I can make this throw a "real" exception though, with a message containing logOffsets.